### PR TITLE
[kube-prometheus-stack] add namespaced roles for Prometheus/Operator

### LINKED ISSUE #4156: [kube-prometheus-stack] Restrict secrets access to namespace
### Is your feature request related to a problem ?

Currently, ClusterRoleBindings are used to attach the ClusterRole to the service account created in the Helm Chart. Read access to the secrets API is configured in the ClusterRole. This allows a potential attacker to read secrets from all namespaces in the Kubernetes cluster.

### Describe the solution you'd like.

Instead of a ClusterRoleBinding, a RoleBinding should be used to only allow access to secrets in the namespace.

### Describe alternatives you've considered.

NONE

### Additional context.

Steps to reproduce:

1. Install Helm Chart in a Kubernetes cluster with `global.rbac.create: true`
2. Create a secret in a different namespace than the one the helm release was installed in
3. Copy kubectl into a running container, e.g. B. grafana.
`kubectl cp kubectl kube-prometheus-stack-grafana-57b9499c74-5jgzl:/tmp -n monitoring -c grafana`
4. Connect to the container the kubectl binary was uploaded to
5. Execute `chmod +x /tmp/kubectl`
6. Show which ServiceAccount is used
`/tmp/kubectl auth whoami`
7. View the secret from a different namespace
`/tmp/kubectl get secrets -n kube-system`

### PR DESCRIPTION
**Pull request by jkroepke**
#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #4156 

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml [hunk 1]
```diff
@@ -1,8 +1,11 @@
 {{- if and .Values.prometheusOperator.enabled .Values.global.rbac.create }}
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
+kind: {{ .Values.prometheusOperator.rbac.namespaced | ternary "Role" "ClusteRole" }}
```
Comment (petenorth) [inline charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml:3]: should `ClusteRole` be `ClusterRole` ?

Comment (jkroepke) [reply charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml:3]: Thats yeah, currently just a brain dump PR

**Comment by QuentinBisson** [comment]
@jkroepke do you feel like you could try to merge main here? I think the PR is quite good :)

**Comment by jkroepke** [comment]
tbh, no idea. half year ago, totally out of context here and limited time to test it. 😅

**Comment by QuentinBisson** [comment]
No worries I'm trying to clean up my list of PRs to review so it's fine if we keep it open :)

**Comment by jkroepke** [comment]
Please take over! I'm happy to take an review over it.

**Comment by yuing32** [comment]
May I know if any other new PR finish this function?

**Pull request closed without merging**